### ISSUE TITLE
FR: allow attaching to PIDs / remote

### ISSUE THREAD

**Issue by GitMensch** [marcsosduma/cobgdb#2]
Depends on #1.

Once this is implemented it would be easy to allow `cobgdb 21321` (attach to pid) or `cobgdb server:port` (attach to gdbserver using extended-remote).

**Comment by marcsosduma** [comment]
Understood, Simon. I'm fixing the code and making some adjustments. However, I'll try to perform this task, meaning considering that the program has already been compiled. My question is how to ensure that it was compiled in the way cobgdb needs it.

**Comment by GitMensch** [comment]
This seems to be implemented (but possibly not on the command line yet).

**Comment by marcsosduma** [comment]
![image](https://github.com/marcsosduma/cobgdb/assets/31162117/76652087-15ce-42d3-8cdc-1bba4b4d3d59)

Hi Simon! For now, it's a command after entering the application. Run cobgdb, insert the breakpoint, and use the "A" command.

**Comment by GitMensch** [comment]
@marcsosduma do we want to close that or keep it open for the command line option?

**Comment by GitMensch** [comment]
Yes, that's useful. I suggest we wait for @eugeniodilo to add that to the documentation (_maybe_ adding two tags: "completed" and "docs-missing" or similar), before closing this issue for good.

**Comment by marcsosduma** [comment]
Hi Simon! Is your suggestion to assume that all the files required for debugging are available? Cobgdb needs the .cob, .c, .c.h, and .c.l.h files in order to perform debugging.

**Comment by GitMensch** [comment]
Yes, that's my suggestion, ideally with the option to tell cobgdb where to look for those.

**Comment by GitMensch** [comment]
Maybe: try to preprocess the C sources specified by GDB, if those aren't available try to create them (with `cobc -c`), otherwise just use them.

**Comment by eugeniodilo** [comment]
> Yes, that's useful. I suggest we wait for [@eugeniodilo](https://github.com/eugeniodilo) to add that to the documentation (_maybe_ adding two tags: "completed" and "docs-missing" or similar), before closing this issue for good.


Unfortunately, I don't understand the purpose of this new `A = ATTACH` function. For this reason, I'm unable to add the function to the documentation.
I'll have to ask Marcos or Simon to explain the purpose of this new function.
Perhaps an example will clarify.
When can and/or should it be used?

**Comment by eugeniodilo** [comment]
Okay, but there's something still unclear.
In both scenarios you described, when you write "_you just open the debugger from a different terminal, attach to the program, debug normally, then either kill or detach,_" 
- what do you mean with "just open the debugger" ?
- what do you mean with "attach to the program" ?
- what do you mean with "kill or detach" ?

I usually open the debugger with the following command:
**cobgdb sourceprog.cob -x -lpdcurses**

In the cases you described in Scenario 1 or 2, do I need to run the following command:
**cobgdb --connect localhost:5555 sourceprog.cob**   ?

So, in what case is the "**A = ATTACH**" command needed ?

**Comment by GitMensch** [comment]
I _guess `cobgdb --connect 1234 sourceprog.cob`  (PID 1234) may also work (haven't tested, also for `--connect` we should not need a source as GDB tells us the source of the current program (if it as compiled with debug info).

The `A` is for the case when you already started `cobgdb` and want to attach from there.
In theory, `A`ttaching to a different PID (or port) later should detach (= program keeps running without debugger).

Note: other than the restart for the attach, the Breakpoints _should_ be kept when re-attaching.
Additional note: the text above is "native GDB's way" - it needs to be tested for cobgdb.
